### PR TITLE
qenv warnings

### PR DESCRIPTION
**Pull request by mhallal1**
related to https://github.com/insightsengineering/teal.code/issues/63

- Added `teal.widgets::verbatim_popup_ui` and `teal.widgets::verbatim_popup_srv` to account for warnings in the modules.
- the button is disabled if there are no warnings.

Please test with example apps.

### CODE REVIEW

File: R/tm_g_bivariate.R [hunk 1]
```diff
@@ -535,6 +538,13 @@ srv_g_bivariate <- function(id,
       width = plot_width
     )
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warnings",
+      reactive(paste(get_warnings(output_q()), collapse = "\n")),
+      title = "Warning",
+      disabled = reactive(is.null(teal.code::get_code(output_q())))
```
Comment (nikolas-burkoff) [inline R/tm_g_bivariate.R:545]: So I do like the button being disabled if there's no warnings - I notice in other modules this isn't there

File: R/tm_g_bivariate.R [hunk 2]
```diff
@@ -370,7 +373,7 @@ srv_g_bivariate <- function(id,
 
     anl_merged_q <- reactive({
       req(anl_merged_input())
-      teal.code::new_qenv(tdata2env(data), code = get_code(data)) %>%
+      teal.code::new_qenv(tdata2env(data), code = teal.data::get_code(data)) %>%
```
Comment (nikolas-burkoff) [inline R/tm_g_bivariate.R:376]: So get_code.tdata lives in teal so do we need `teal.data::` here? Would be really sad if we do...

File: R/tm_g_bivariate.R (hunk 1 above)
Comment (nikolas-burkoff) [inline R/tm_g_bivariate.R:545]:
```suggestion
      disabled = reactive(is.null(teal.code::get_warnings(output_q())))
```

File: R/tm_g_bivariate.R (hunk 2 above)
Comment (mhallal1) [reply R/tm_g_bivariate.R]: https://github.com/insightsengineering/teal/pull/739

File: R/tm_g_scatterplot.R [hunk 3]
```diff
@@ -869,6 +872,12 @@ srv_g_scatterplot <- function(id,
       }
     })
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warning",
+      verbatim_content = reactive(get_warnings(output_q())),
```
Comment (nikolas-burkoff) [inline R/tm_g_scatterplot.R:877]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(output_q())),
```

File: R/tm_outliers.R [hunk 4]
```diff
@@ -1078,6 +1081,12 @@ srv_outliers <- function(id, data, reporter, filter_panel_api, outlier_var,
       }
     })
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warning",
+      verbatim_content = reactive(get_warnings(final_q())),
```
Comment (nikolas-burkoff) [inline R/tm_outliers.R:1086]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(final_q())),
```

File: R/tm_g_bivariate.R [hunk 5]
```diff
@@ -329,7 +329,10 @@ ui_g_bivariate <- function(id, ...) {
         )
       )
     ),
-    forms = teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code"),
+    forms = tagList(
+      teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code"),
+      teal.widgets::verbatim_popup_ui("warnings", button_label = "Show Warnings")
```
Comment (nikolas-burkoff) [inline R/tm_g_bivariate.R:334]:
```suggestion
      teal.widgets::verbatim_popup_ui(ns("warnings"), button_label = "Show Warnings")
```

File: R/tm_g_bivariate.R [hunk 6]
```diff
@@ -535,6 +540,12 @@ srv_g_bivariate <- function(id,
       width = plot_width
     )
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warnings",
+      verbatim_content = reactive(get_warnings(output_q())),
```
Comment (nikolas-burkoff) [inline R/tm_g_bivariate.R:545]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(output_q())),
```

File: R/tm_g_distribution.R [hunk 7]
```diff
@@ -316,7 +316,10 @@ ui_distribution <- function(id, ...) {
         )
       )
     ),
-    forms = teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code"),
+    forms = tagList(
+      teal.widgets::verbatim_popup_ui(ns("warning"), "Show Warnings"),
```
Comment (nikolas-burkoff) [inline R/tm_g_distribution.R:320]: Some are id `warnings` others `warning`

File: R/tm_g_response.R [hunk 8]
```diff
@@ -333,7 +336,7 @@ srv_g_response <- function(id,
       plot_call <- substitute(
         expr =
           ggplot(ANL2, aes(x = x_cl, y = ns)) +
-            geom_bar(aes(fill = resp_cl), stat = "identity", position = arg_position),
+            geom_bar(aes(fill = resp_cl), stat = "identity", position = arg_position, dd = ""),
```
Comment (nikolas-burkoff) [inline R/tm_g_response.R:339]: Is this right?

File: R/tm_g_response.R [hunk 9]
```diff
@@ -429,9 +432,15 @@ srv_g_response <- function(id,
       width = plot_width
     )
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warning",
+      verbatim_content = reactive(get_warnings(output_q())),
```
Comment (nikolas-burkoff) [inline R/tm_g_response.R:437]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(output_q())),
```

File: R/tm_g_response.R [hunk 10]
```diff
@@ -429,9 +432,15 @@ srv_g_response <- function(id,
...
+      title = "Warning"
+    )
+
     teal.widgets::verbatim_popup_srv(
       id = "rcode",
-      verbatim_content = reactive(teal.code::get_code(output_q())),
+      verbatim_content = reactive(get_code(output_q())),
```
Comment (nikolas-burkoff) [inline R/tm_g_response.R:443]:
```suggestion
      verbatim_content = reactive(teal.code::get_code(output_q())),
```

File: R/tm_t_crosstable.R [hunk 11]
```diff
@@ -315,6 +318,12 @@ srv_t_crosstable <- function(id, data, reporter, filter_panel_api, label, x, y,
       table_r = table_r
     )
 
+    teal.widgets::verbatim_popup_srv(
+      id = "warning",
+      verbatim_content = reactive(get_warnings(output_q())),
```
Comment (nikolas-burkoff) [inline R/tm_t_crosstable.R:323]:
```suggestion
      verbatim_content = reactive(teal.code::get_warnings(output_q())),
```

File: R/tm_g_response.R (hunk 8 above)
Comment (mhallal1) [reply R/tm_g_response.R:339]: i was testing with this and i forgot it there, will remove.

File: R/tm_g_distribution.R (hunk 7 above)
Comment (mhallal1) [reply R/tm_g_distribution.R:320]: fixed

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**